modules/@apostrophecms/page/index.js

```diff
@@ -1391,10 +1401,13 @@ database.`);
           await self.emit('serve', req);
           await self.serveNotFound(req);
         } catch (err) {
-          await self.serveDeliver(req, err);
-          return;
+          return await self.serve500Error(req, err);
```

Comment:
I ran into this while trying to debug my work on `data.localizations`: we were running 500 error pages through the full stack, including sending out `beforeSend` events, which was not our intent because it is too likely to produce the very same error. The actual templateError template is designed to be self-sufficient, so I altered the logic to take advantage of that.